### DISCUSSION TITLE
Properties detail are missing from VertexLabel using JanusGraphManagement API.

### DISCUSSION THREAD

**Discussion by dishabehrani77** [JanusGraph/janusgraph#4297] [Q&A] [answered]
I am using janugraph-1.0.0 and cassandra as backend storage.This is configuration file with cassandra details,  janusgraph-cql/.properties 
![image](https://github.com/JanusGraph/janusgraph/assets/162420429/f448a0b3-5962-47a5-a94f-d696d407f1a6)

 
Now using gremlin console, I have added vertex and I am able to fetch it .
![image](https://github.com/JanusGraph/janusgraph/assets/162420429/ab0da8b9-2d19-4d6a-8552-5ec3808115a0)

 

Now on trying to get the properties and the datatype, using JanusGraphManagemnet API. 
![image](https://github.com/JanusGraph/janusgraph/assets/162420429/67d724a4-a157-4963-b3a6-65fccf307c1b)

 
I am unable to get the property details , Is there a away to get the details using management API ?

**Comment by FlorianHockmann** [comment, marked as answer]
You can print the full schema via `mgmt.printSchema()`. That should also display the property keys.

However, `mgmt.getVertexLabel().keys([...])`, `mgmt.getVertexLabel([...]).properties()`, and `mgmt.getVertexLabel([...]).mappedProperties()` don't show any properties, because JanusGraph defines property keys globally and not per vertex label.
You can get all property keys via `mgmt.printPropertyKeys()`.

If you want to only allow certain property keys on certain labels, then you can use [schema constraints](https://docs.janusgraph.org/schema/#schema-constraints). These are not enabled by default so you have to enable them yourself.
If you use schema constraints, then you can actually find the allowed property keys for a given vertex label via `mgmt.getVertexLabel([...]).mappedProperties()`.

`keys()` and `properties()` still won't work. They are only available in the API because JanusGraph stores schema information internally also as a graph and the JanusGraph vertices have these methods.
